cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/FlowletTwillRunnable.java

```diff
@@ -30,4 +38,18 @@
   protected Class<FlowletProgramRunner> getProgramClass() {
     return FlowletProgramRunner.class;
   }
+
+  @Override
+  public void handleCommand(Command command) throws Exception {
+    try {
+      super.handleCommand(command);
+    } catch (ExecutionException e) {
+      Throwable rootCause = Throwables.getRootCause(e);
+      if (!"Suspension not allowed".equals(rootCause.getMessage())
```

Comment:
That's not a very great fix, but a quick fix. The IllegalStateException is thrown from the AbstractProgramController when the controller state cannot transit to suspend/resume state. Unfortunately, there are other failure that also throw IllegalStateException which we don't want to ignore. Any better suggestion?